controller/recurring_job_controller.go

```diff
@@ -220,6 +224,51 @@ func (control *RecurringJobController) syncRecurringJob(key string) (err error)
 	return nil
 }
 
+func (control *RecurringJobController) cleanupVolumeRecurringJob(recurringJob *longhorn.RecurringJob) error {
+	// check if other recurring jobs contains same groups
+	checkRecurringJobs, err := control.ds.ListRecurringJobs()
+	if err != nil {
+		return err
+	}
+	rmGroups := []string{}
+	for _, group := range recurringJob.Spec.Groups {
+		inUse := false
+		for _, checkRecurringJob := range checkRecurringJobs {
+			if checkRecurringJob.Name == recurringJob.Name {
+				continue
+			}
+			if util.Contains(checkRecurringJob.Spec.Groups, group) {
+				inUse = true
+				break
+			}
+		}
+		if !inUse {
+			rmGroups = append(rmGroups, group)
+		}
+	}
+
+	// delete volume labels
+	volumes, err := control.ds.ListVolumes()
```

Comment:
We need to check each volume and each labels in volumes to delete the labels for the enqueued recurring job. Also need to check if the recurring job is the last in the group. I prefer to keep it this way because to separate the logic into two functions means it needs to loop twice through the labels to get the names (job & job-group).

Reply:
OK. In brief, you prefer to use a single loop here.